core/src/robot/FixedFrameConstraint.cc

```diff
@@ -0,0 +1,65 @@
+#include "jiminy/core/robot/Robot.h"
+#include "jiminy/core/Utilities.h"
+
+#include "jiminy/core/robot/FixedFrameConstraint.h"
+
+
+namespace jiminy
+{
+    FixedFrameConstraint::FixedFrameConstraint(std::string const & frameName) :
+    AbstractConstraint(),
+    frameName_(frameName),
+    frameId_(0)
+    {
+        // Empty on purpose
+    }
+    FixedFrameConstraint::~FixedFrameConstraint(void)
+    {
+        // Empty on purpose
+    }
+
+    hresult_t FixedFrameConstraint::initialize(Model *model)
+    {
+        model_ = model;
+        // Refresh proxies: this checks for the existence of frameName_ in model_.
+        hresult_t returnCode = refreshProxies();
+        isInitialized_ = returnCode == hresult_t::SUCCESS;
+        return returnCode;
+    }
+
+    hresult_t FixedFrameConstraint::refreshProxies()
+    {
+        return getFrameIdx(model_->pncModel_, frameName_, frameId_);
+    }
+
+    matrixN_t FixedFrameConstraint::getJacobian(vectorN_t const & q) const
```

Comment:
As I said, return by const ref

Reply:
const ref doesn't work as is with the python bindings.

```
ue_policy_to_wrap_functions_returning<const Eigen::Matrix<double, -1, 1>&>) (const Eigen::Matrix<double, -1, 1>&)’
     return rc( (tc().*f)(BOOST_PP_ENUM_BINARY_PARAMS_Z(1, N, ac, () BOOST_PP_INTERCEPT)) );
```

Reply:
yes, you have to do something like this:
```
                .add_property("is_initialized", bp::make_function(&Engine::getIsInitialized,
                                                bp::return_value_policy<bp::copy_const_reference>()))
```